#!/usr/bin/env python3

from dynamic_reconfigure.parameter_generator_catkin import *
PACKAGE = "movement_pipeline"

gen = ParameterGenerator()

gen.add("ArmLiftJoint", double_t, 0, "Arm Lift Joint (linear in z) [m]", 0.0, 0.0, 0.69)
gen.add("ArmFlexJoint", double_t, 0, "Arm Flex Joint (rotation in -y) [rad]", 0.0, -2.617, 0.0)
gen.add("ArmRollJoint", double_t, 0, "Arm Roll Joint (rotation in z) [rad]", -1.57079632679, -1.919, 3.665)
gen.add("WristFlexJoint", double_t, 0, "Wrist Flex Joint (rotation in -y) [rad]", -1.57079632679, -1.919, 1.221)
gen.add("WristRollJoint", double_t, 0, "Wrist Roll Joint (rotation in z) [rad]", 0.0, -1.919, 3.665)
gen.add("HeadPanJoint", double_t, 0, "Head Pan Joint (rotation in z) [rad]", 0.0, -3.839, 1.745)
gen.add("HeadTiltJoint", double_t, 0, "Head Tilt Joint (rotation in -y) [rad]", 0.0, -1.570, 0.523)
gen.add("Gripper", double_t, 0, "Gripper [rad]", 1.0, 0.0, 1.2)

gen.add("Line", str_t, 0, "String for the HSR to say", "Default")

gen.add("Table", bool_t, 0, "Resets all joints and moves to the table", False)
gen.add("Reset", bool_t, 0,"Reset all values", False)


exit(gen.generate(PACKAGE, "movement_pipeline", "JointMovement"))
